src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -2738,6 +2738,16 @@ internal BoundExpression GenerateConversionForAssignment(TypeSymbol targetType,
                 }
             }
 
+            if (operand.Kind == BoundKind.DefaultPendingInference)
+            {
+                // PROTOTYPE(default) Need to handle type parameters here (`where T : struct` for instance)
+                if (!targetType.IsStructType() || targetType.IsNullableType())
+                {
```

Comment:
Do you intend that default() won't work for enum types?

---
In reply to: [77566856](https://github.com/dotnet/roslyn/pull/13603#discussion_r77566856) [](ancestors = 77566856)